applications/HDF5Application/custom_io/hdf5_model_part_io.cpp

```diff
@@ -197,7 +197,9 @@ void ModelPartIO::WriteModelPart(ModelPart& rModelPart)
     Internals::WriteBufferSize(*mpFile, mPrefix, rModelPart.GetBufferSize());
     WriteProperties(rModelPart.rProperties());
     Internals::WriteDataValueContainer(*mpFile, mPrefix + "/ProcessInfo", rModelPart.GetProcessInfo());
-    rModelPart.Nodes().Sort(); // Avoid inadvertently reordering partway through
+
+    // Charlie: A write function should not modify the model part. This is a bug. If its broken needs to be fixed elsewhere.
+    // rModelPart.Nodes().Sort(); // Avoid inadvertently reordering partway through
```

Comment:
@sunethwarna I think this fix is yours, we need to find another solution o call the sort outside this function.